src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -283,27 +336,19 @@ def _get_imports_info(
     address: Address,
     owners_per_import: Iterable[PythonModuleOwners],
     parsed_imports: ParsedPythonImports,
-    explicitly_provided_deps: ExplicitlyProvidedDependencies,
 ) -> dict[str, ImportResolveResult]:
     def _resolve_single_import(owners, import_name) -> ImportResolveResult:
         if owners.unambiguous:
             return ImportResolveResult(ImportOwnerStatus.unambiguous, owners.unambiguous)
 
-        explicitly_provided_deps.maybe_warn_of_ambiguous_dependency_inference(
-            owners.ambiguous,
-            address,
-            import_reference="module",
-            context=f"The target {address} imports `{import_name}`",
-        )
-        maybe_disambiguated = explicitly_provided_deps.disambiguated(owners.ambiguous)
-        if maybe_disambiguated:
-            return ImportResolveResult(ImportOwnerStatus.disambiguated, (maybe_disambiguated,))
-        elif import_name.split(".")[0] in DEFAULT_UNOWNED_DEPENDENCIES:
+        if import_name.split(".")[0] in DEFAULT_UNOWNED_DEPENDENCIES:
             return ImportResolveResult(ImportOwnerStatus.unownable)
         elif parsed_imports[import_name].weak:
-            return ImportResolveResult(ImportOwnerStatus.weak_ignore)
+            # what is weak_ignore? should we try to disambiguate these?
+            return ImportResolveResult(ImportOwnerStatus.weak_ignore, owners.ambiguous)
```

Comment:
the import detector will flag imports as weak if they won't obviously fail module loading. For example:
```python
import mypackage
```
will fail the module load if "mypackage" isn't importable, but 
```python
try:
    import mypackage
    has_mypackage = True
except ModuleNotFoundError:
    has_mypackage = False
```
is a pattern for optionally using "mypackage". Since it is valid for "mypackage" to not be inferable, the detector marks it as a weak import. If we don't resolve the import here, we flag it as `weak_ignore` so that Pants can surface to a user that it has detected this import, has not found the thing it imports, and is ok with that. (This was the original motivation for me adding the debug feature.)

Reply:
I think you're correct that it is necessary to disambiguate weak ignores. Thinking about the whole "owner resolver" action, there are 3 things we do:
- find an unambiguous owner
- identify an unambiguous owner from ambiguous owners based on hints (explicit deps, visibility rules)
- make excuses for why we can't find a unique owner
`weak_ignore` is definitely an excuse, so I think we have to attempt ambiguity resolution before we can say that for certain.

I think a more straightforward indication is that there are `owners.ambiguous` to pass on.